tests/functional/monitoring/prometheus/alerts/test_alert_mds_cache_high_usage.py

```diff
@@ -0,0 +1,211 @@
+        pod_obj = dc_pod_factory(
+            pvc=pvc_obj, access_mode=access_mode, interface=interface
+        )
+        # Copy meta_data_io.py to fedora pod
+        log.info("Copying meta_data_io.py to fedora pod ")
+        cmd = f"oc cp {file} {pvc_obj.namespace}/{pod_obj.name}:/"
+        helpers.run_cmd(cmd=cmd)
+        log.info("meta_data_io.py copied successfully ")
+
+        # Run meta_data_io.py on fedora pod
+        log.info("Running meta data IO on fedora pod ")
+        metaio_executor = ThreadPoolExecutor(max_workers=1)
+        # self.metaio_thread = metaio_executor.submit(
+        metaio_executor.submit(
+            pod_obj.exec_sh_cmd_on_pod, command="python3 meta_data_io.py"
+        )
+
+
+def active_mds_alert_values(threading_lock):
+    active_mds = cluster.get_active_mds_info()["mds_daemon"]
+    sr_mds = cluster.get_mds_standby_replay_info()["mds_daemon"]
+    cache_alert = constants.ALERT_MDSCACHEUSAGEHIGH
+    message = f"High MDS cache usage for the daemon mds.{active_mds}."
+    description = (
+        f"MDS cache usage for the daemon mds.{active_mds} has exceeded above 95% of the requested value."
+        f" Increase the memory request for mds.{active_mds} pod."
+    )
+    runbook = (
+        "https://github.com/openshift/runbooks/blob/master/alerts/"
+        "openshift-container-storage-operator/CephMdsCacheUsageHigh.md"
+    )
+    state = "firing"
+    severity = "critical"
+
+    api = prometheus.PrometheusAPI(threading_lock=threading_lock)
+    alerts_response = api.get("alerts", payload={"silenced": False, "inhibited": False})
+    prometheus_alerts = alerts_response.json()["data"]["alerts"]
+
+    prometheus.verify_mds_alerts(
+        alert_name=cache_alert,
+        msg=message,
+        description=description,
+        runbook=runbook,
+        state=state,
+        severity=severity,
+        alerts=prometheus_alerts,
+        active_mds=active_mds,
+        standby_mds=sr_mds,
+    )
+    log.info("Alert verified successfully")
+    return True
+
+
+@tier2
+@blue_squad
+class TestMdsMemoryAlerts:
+    @pytest.mark.polarion_id("OCS-5570")
+    def test_alert_triggered(self, run_metadata_io_with_cephfs, threading_lock):
+        # This function verifies the mds cache alert triggered or not
+        log.info(
+            "Metadata IO started in the background. Script will sleep for 15 minutes before validating the MDS alert"
+        )
+        time.sleep(900)
+        log.info("Validating the alert now")
+        assert active_mds_alert_values(threading_lock)
+
+    @pytest.mark.polarion_id("OCS-5571")
+    def test_mds_cache_alert_with_active_node_drain(
+        self, run_metadata_io_with_cephfs, threading_lock
+    ):
+        # This function verifies the mds cache alert with active mds running node drain
+        log.info(
+            "Metadata IO started in the background. Lets wait for 15 minutes before validating the MDS alert"
+        )
+        time.sleep(900)
```

Comment:
Not critical.

do you want to pull these timeouts to module or class variables? 
I see in PR 900, 600, 100 sec. This may help to maintain the module.

Reply:
Good suggestion, implemented the same.